tenseal/cpp/tensors/ckkstensor.cpp

```diff
@@ -0,0 +1,185 @@
+#include "tenseal/cpp/tensors/ckkstensor.h"
+
+namespace tenseal {
+
+using namespace seal;
+using namespace std;
+
+CKKSTensor::CKKSTensor(const shared_ptr<TenSEALContext>& ctx,
+                       const PlainTensor<double>& tensor,
+                       std::optional<double> scale)
+    : _shape(tensor.shape()), _strides(tensor.strides()) {
+    this->link_tenseal_context(ctx);
+    if (scale.has_value()) {
+        this->_init_scale = scale.value();
+    } else {
+        this->_init_scale = ctx->global_scale();
+    }
+
+    vector<double> plain_data = tensor.data();
+    for (int i = 0; i < plain_data.size(); i++)
+        _data.push_back(
+            CKKSTensor::encrypt(ctx, this->_init_scale, plain_data[i]));
```

Comment:
It's for another PR, but I think it could be useful to add iterator support for the PlainTensors.

Reply:
Yeah, that would be great!